### PR TITLE
**Breaking** [Alert] Introduce notice type, remove mode prop, and title slot will always show if it has content

### PR DESCRIPTION
**Pull request by petemill**
If a title slot was provided and the mode wasn't set to full then the title would never show. Seems to me that the full mode is only for adjustments made to make the alert look good with a title - ~so this PR forces 'full' mode when a title is present (or hasTitle prop is true for when the framework can't have an optional slot!)~. After discussion with @aguscruiz this PR removes `mode="full"` (leaving only a single mode, and removing it) as it wasn't being used by any designs, and allows title without any mode restriction

This also adds a variant for 'notice' type, which is color-less and has smaller features.

image
image

### CODE REVIEW

File: src/components/alert/alert.svelte [hunk 1]
```diff
@@ -21,20 +24,18 @@
   export let isToast = false
   export let hasActions = $$slots.actions
   export let hasContentAfter = $$slots['content-after']
+  export let hasTitle = $$slots.title
```
Comment (fallaciousreasoning) [inline src/components/alert/alert.svelte:27]: Hey @petemill what's the point of exporting this? I feel like it also might be slightly confusing to set `mode=simple` and provide a title and have the alert show in `full` mode?

I think at least we shouldn't export this :smile:

Additionally, I think `showTitle` is probably a better name, as we hide the title when this is false (and we might have a title, and just decide not to show it)

File: src/components/alert/alert.svelte [hunk 2]
```diff
@@ -44,7 +45,7 @@
   </div>
   {/if}
   <div class="content">
-    {#if mode === 'full' && $$slots.title}
+    {#if hasTitle && $$slots.title}
```
Comment (fallaciousreasoning) [inline src/components/alert/alert.svelte:48]: Are we doing this for Svelte, where you can't conditionally render a slot?

Comment (AlanBreck) [reply src/components/alert/alert.svelte:48]: Guessing yes, and that it's the same reason for [exporting `hasTitle`](https://github.com/brave/leo/pull/912#discussion_r1855581102).

Comment (fallaciousreasoning) [reply src/components/alert/alert.svelte:48]: Are there cases where we're going to conditionally show a title though? I feel like that's generally decided upfront

File: src/components/alert/alert.svelte (hunk 1 above)
Comment (petemill) [reply src/components/alert/alert.svelte:27]: Well yes I was confused as to why we are exporting `hasActions` and `hasContentAfter` and guessed it was for svelte (and storybook), in order to hide the title element when no title is provided.

**Review by fallaciousreasoning — APPROVED** [review]
Happy to land this as-is 'cause its the same pattern as everything else :shrug:

File: src/components/alert/alert.svelte (hunk 2 above)
Comment (AlanBreck) [reply src/components/alert/alert.svelte:48]: > Are there cases where we're going to conditionally show a title though? I feel like that's generally decided upfront

Not that I'm aware of for `title`.

Comment (petemill) [reply src/components/alert/alert.svelte:48]: I see what you're saying - the svelte issue is about having slotted content in a templated conditional. In any case, I just tested what happens when we render the empty title element and there are no visual issues, so I might as well let it get rendered...

**Review by fallaciousreasoning — APPROVED** [review]
Can you mark this PR as **BREAKING** so we know in the changelog?

File: src/components/alert/alert.svelte [hunk 3]
```diff
@@ -1,40 +1,35 @@
...
+    'success': 'check-circle-filled',
+    'notice': 'info-outline'
   }
 </script>
 
 <script lang="ts">
   import Icon from '../icon/icon.svelte'
 
   export let type: AlertType = 'error'
-  export let mode: AlertMode = 'simple'
```
[the remark this answers is not in the document]
Comment (petemill) [reply src/components/alert/alert.svelte:20]: there is no concept of `full` - there is no longer an icon size difference

**Comment by AlanBreck** [comment]
> Can you mark this PR as BREAKING so we know in the changelog?

What about it will be breaking?

**Comment by petemill** [comment]
> > Can you mark this PR as BREAKING so we know in the changelog?
> 
> What about it will be breaking?

If any Alerts were using `mode="full"` it will cause an error?

**Pull request merged**